client/lib/analytics/with-tracking-tool/index.jsx

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { loadTrackingTool } from 'state/analytics/actions';
+
+export default trackingTool => EnhancedComponent => {
+	class WithTrackingTool extends Component {
+		static displayName = `WithTrackingTool( ${ EnhancedComponent.displayName ||
```

Comment:
Should this default to something in case `displayName` or `name` are not available?

Reply:
Yup, and we've been doing that for other HoCs too, it's not a new pattern 😉

Reply:
(Actually defaults to `WithTrackingTool(undefined)` then.)

Reply:
🤷‍♂️ TBH I think we'll be loading this for container components, which are more likely to have names. Anyway, if you have a recommendation here, I'm open to refactor.

Reply:
I think it's okay as is and if any changes is needed, I'd add `|| ''`

Reply:
👍 I can follow-up with that.

Reply:
Addressed in #30844